js/plugins/google-cloud/src/telemetry/action.ts

```diff
@@ -55,35 +55,37 @@ class ActionTelemetry implements Telemetry {
 
     const actionName = (attributes['genkit:name'] as string) || '<unknown>';
     const path = (attributes['genkit:path'] as string) || '<unknown>';
-    const flowName =
-      (attributes['genkit:metadata:flow:name'] as string) ||
-      extractOuterFlowNameFromPath(path);
+    let featureName = (attributes['genkit:metadata:flow:name'] ||
+      extractOuterFeatureNameFromPath(path)) as string;
+    if (featureName === '<unknown>') {
+      featureName = actionName;
+    }
     const state = attributes['genkit:state'] || 'success';
     const latencyMs = hrTimeToMilliseconds(
       hrTimeDuration(span.startTime, span.endTime)
     );
     const errorName = extractErrorName(span.events);
 
     if (state === 'success') {
-      this.writeSuccess(actionName, flowName, path, latencyMs);
+      this.writeSuccess(actionName, featureName, path, latencyMs);
       return;
     }
     if (state === 'error') {
-      this.writeFailure(actionName, flowName, path, latencyMs, errorName);
+      this.writeFailure(actionName, featureName, path, latencyMs, errorName);
     }
 
     logger.warn(`Unknown action state; ${state}`);
   }
 
   private writeSuccess(
     actionName: string,
-    flowName: string,
+    featureName: string,
```

Comment:
Are you sure we should switch from flowName to featureName?

Since we're adding feature-level metrics without removing flow-level metrics, why not include featureName as an additional dimension on the action metrics instead of replacing?

Reply:
I guess that's true, `flowName` can still be set if this action is part of a feature that is also a flow. But the question is, will `flowName` be used in the future, or will it be more "feature paths" (to drive things like our span filter)?

Reply:
Feature name is the same as how flow name worked before, if you are in a flow.

Reply:
But flow and feature may be different if the top-most Genkit span isn't a flow.  If we're supporting separate metrics for these concepts, why wouldn't we keep them as separate dimensions on other metrics?

Reply:
DOne